File: selfstudy/golang/apples_sum.go
/*
task 201:
Apples - Sum
You need to calculate the total sum of apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func TotalApples(apples []int) int {
	sum := 0
	for _, v := range apples {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, TotalApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, TotalApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, TotalApples([]int{1, 2, 3})},
		{[]int{100, 200}, TotalApples([]int{100, 200})},
		{[]int{}, TotalApples([]int{})},
	}
	
	for idx, t := range tests {
		res := TotalApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think

File: selfstudy/python/apples_sum.py
'''
task 101:
Apples - Sum
You need to calculate the total sum of apple weights in grams
'''

def total_apples(apples):
    return sum(apples)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], total_apples([10, 20, 30])),
        ([50, 60, 70], total_apples([50, 60, 70])),
        ([1, 2, 3], total_apples([1, 2, 3])),
        ([100, 200], total_apples([100, 200])),
        ([], total_apples([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = total_apples(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think
